lib/logstash/outputs/file.rb

```diff
@@ -141,7 +142,7 @@ def multi_receive_encoded(events_and_encoded)
           # append to the file
           chunks.each {|chunk| fd.write(chunk) }
         end
-        flush(fd)
+        flush(fd) unless @flusher && @flusher.alive?
```

Comment:
now that we rely on Interval for periodic flushing, I suggest either consistently using `flush(fd)` everywhere, or just just `fd.flush`